incubator/@react-native-webapis/scan/src/ignored_dirs.rs

```diff
@@ -0,0 +1,115 @@
+/**
+ * THIS ARRAY MUST BE SORTED! We perform binary searches on it.
+ *
+ * List of directories to ignore.
+ */
+pub static IGNORED_DIRECTORIES: &'static [&str] = &[
+    "@babel",
+    "@changesets",
+    "@eslint-community",
+    "@haul-bundler",
+    "@jest",
+    "@pnpm",
+    "@react-native",
+    "@react-native-community",
+    "@react-native-webapis",
+    "@react-native-windows",
```

Comment:
```suggestion
    "@react-native-community",
    "@react-native-mac",
    "@react-native-webapis",
    "@react-native-windows",
```

if I'm understanding correctly, we need to add the new "@react-native-mac" entry here since it's what the RNmacos team is using for their monorepo (post 72 monorepo landing).

The tool was indeed annoyed at some files in it:
<img width="945" alt="Screenshot 2023-09-19 at 15 20 30" src="https://github.com/microsoft/rnx-kit/assets/16104054/43126cdd-8309-4123-9c95-c697b34f2905">

(I'm not sure what we want/should do about the `Failed to parse source file: ./packages/typescript-service/test/__fixtures__/sample-snapshot-file.js` error)

Reply:
These are usually due to the source file being Flow typed. Ideally, we should strip out the types and try again. But it doesn't skew the numbers that much, so I will address it later.